source/slang/slang-ir-spirv-legalize.cpp

```diff
@@ -1612,6 +1612,41 @@ struct SPIRVLegalizationContext : public SourceEmitterBase
         }
     }
 
+    void legalizeSPIRVLinkageFunc(IRFunc* func)
+    {
+        // Check the `HLSLExport` decoration or `DownstreamExport` decoration. If any performs, we
+        // assume this function is exported, add SpirV export decoration and add linkage attribute
+        // later.
+        auto hlslExportDecor = func->findDecoration<IRHLSLExportDecoration>();
```

Comment:
I was not involved in the prior reviews of this PR, so I am a bit confused why we would want to take a seemingly HLSL/D3D-specific decoration and use it to infer that we should add a SPIR-V-specific decoration.

If the reality is that `IRHLSLExportDecoration` is in practice representing something more general, then I would prefer that either this PR include a rename of that decoration to something more appropriate, or such a rename be introduced via a separate PR.

(I realize my feedback might contradict with Yong's, so I'll probably need to go look through the earlier review comments at some point to understand the full context)